sources/__tests__/accessible-name.js

```diff
@@ -191,6 +200,52 @@ describe("to upstream", () => {
 	});
 });
 
+describe("slots", () => {
+	beforeAll(() => {
+		customElements.define(
+			"custom-button",
+			class extends HTMLElement {
+				constructor() {
+					super();
+					const shadowRoot = this.attachShadow({ mode: "open" });
+					shadowRoot.innerHTML = `<button data-test><slot></slot></button>`;
+				}
+			}
+		);
+
+		customElements.define(
+			"custom-button-with-default",
+			class extends HTMLElement {
+				constructor() {
+					super();
+					const shadowRoot = this.attachShadow({ mode: "open" });
+					shadowRoot.innerHTML = `<button data-test><slot>Default name</slot></button>`;
```

Comment:
Should this `data-test` attribute be here? I guess it's harmless right now but might be confusing if an actual test forgets to add a `data-test` attribute.

Reply:
This is for the `testShadowDomMarkup` function, which queries for an element in the shadowRoot with the data-test attribute. Is there another way you'd prefer to identify the element under test?

Reply:
Is this actually how a screen reader works? The idea for `data-test` was that that would be the element for which a screen reader is determining the accessible name. 

In other words: Say I have a `CustomButton` for which I write test: How would I query for the element for which I want the accessible name? You wouldn't put `data-test` inside the implementation but rather `<custom-button data-test />` and compute the accessible name for that.

Reply:
So I guess this all makes sense (barely any ShadowDOM knowledge). My main issue is that `testShadoDOMMarkup` is tied to the custom elements inside the `"slots"` block. If you could move `testShadowDOMMarkup` inside the same block then we're probably not leaking implementation details.

My concern is not with the current tests but how these evolve over time. From my experience test utilities get oftentimes abused so we need to make sure they're scoped to their actual usage when introduced.

Reply:
I think there's some confusion as to which element's name we're determining. We're not determining the name of `<custom-button>`, but the `<button>` inside of `<custom-button>`. Semantically, `<custom-button>` is equivalent to a `<span>` -- it has no special meaning to assistive tech. So, that's why I put `data-test` on the inner `<button>` -- that's the element under test.

I think re-using the `data-test` is confusing because it implies that the `<custom-button>` is under test. What do you think about introducing a `data-root` attribute to query for the custom element container and leaving `data-test` on the `<button>` we're actually testing?

So our test fixture would be
```html
<custom-button data-root></custom-button>
```

And our test helper would be:
```diff
function testShadowDomMarkup(markup, accessibleName) {
	const container = renderIntoDocument(markup);

	const testNode = container
-		.querySelector("[data-test]")
+		.querySelector("[data-root]")
		.shadowRoot.querySelector("[data-test]");
	expect(testNode).toHaveAccessibleName(accessibleName);
}

Reply:
> <custom-button data-root></custom-button>

That makes the most sense to me :+1:

Reply:
Tests have been updated to use `data-root` on the custom element containing the element under test. Please resolve if you're satisfied with the change.